ckanext/datastore/config_declaration.yaml

```diff
@@ -92,6 +92,16 @@ groups:
       them. It can be overwritten by the user by passing the "lang" parameter to
       "datastore_search" and "datastore_create".
 
+  - default: 1_000
+    key: ckan.datastore.default_query_timeout_ms
+    type: int
+    example: 180_000
+    description: >
+      The default query timeout (in milliseconds) for the datastore queries.
+      This is used when executing queries against the datastore.
+      "search"/"search_sql", "upsert", and "create" actions will use this
```

Comment:
It's worth having a longer timeout for `upsert` and `create`. Would you add config options for these as well? These can default to 60s.

I might have been too aggressive with my 1s default suggestion for the search queries. To prevent too many reports of failures for large sites let's set that to 5s.

Reply:
Fixed here @wardi: https://github.com/avdata99/ckan/commit/9dee07417ceee7b444d543ff540dd9175b0f2dca